server/src/main/java/org/opensearch/common/util/concurrent/ThreadContext.java

```diff
@@ -483,6 +483,18 @@ public void addResponseHeader(final String key, final String value) {
         addResponseHeader(key, value, v -> v);
     }
 
+    /**
+     * Remove the {@code value} for the specified {@code key}.
+     *
+     * @param key         the header name
+     */
+    public void removeResponseHeader(final String key) {
+        ThreadContextStruct threadContextStruct = threadLocal.get();
+        if (threadContextStruct.responseHeaders != null) {
```

Comment:
It will be better to document the cases where the `responseHeaders` might be null and why ?